src/System.Management.Automation/engine/InternalCommands.cs

```diff
@@ -439,16 +501,26 @@ protected override void ProcessRecord()
                                         // PipelineStoppedException can be caused by select-object
                                         throw;
                                     }
-                                    catch (Exception)
+                                    catch (Exception ex)
                                     {
-                                        // When the property is not gettable or it throws an exception.
-                                        // e.g. when trying to access an assembly's location property, since dynamic assemblies are not backed up by a file,
-                                        // an exception will be thrown when accessing its location property. In this case, return null.
-                                        WriteObject(null);
+                                        if (isBlindDynamicAccess)
+                                        {
+                                            errorRecord = new ErrorRecord(ex,
```

Comment:
I know you're not a fan of silent errors, but I think consistency here is important - you wouldn't want an error just because the object implemented `IDMOP`, e.g. if some object didn't in the past, but a new version does, scripts would break when they silently "worked" before.

Reply:
The problem here is that I can't tell the difference between "it failed because the property does not exist" (let's call this case 1) and "it failed because accessing it actually threw some exception" (let's call that case 2).

Existing PowerShell behavior is different for these two cases: case 1 gets an error (which is possible because the ETS tells us up front if the property exists or not), and case 2 does not.

For non-IDMOP objects, there's no problem; but for IDMOPs, we have the chance to attempt a "blind" access, but the cost is that we must have the same response to both cases (because we cannot distinguish between the two). So we have to make a choice: we can either swallow ALL errors (including "The property 'Blarg' does not exist"), or expose them all.  

I felt like it was more important to preserve the behavior of showing "The property 'Blarg' does not exist" (case 1) errors than to suppress "FooException thrown when accessing Bloop property" (case 2) errors. This decision certainly must be influenced by my predisposition against silent errors, but actually I was going for usability: a typo or otherwise trying to access a property that does not exist is a common mistake, and generally easily fixed when presented with the error, and if I had to pick one of "error suppression" or "tell the user s/he goofed", I'd choose the latter.

What do you think?

Reply:
(I can at least add a comment about this)

Reply:
You can make some assumptions though:
* non-dynamic members must exist
* if `GetDynamicMemberNames` says the member exists, trust it

Reply:
Yes--in fact, those are exactly the assumptions I am making: if it had shown up in `GetDynamicMemberNames`, it would show up as a dynamic member, and `isBlindDynamicAccess` would be false. (And it would be false for normal properties as well, of course.) I'll add more comments about this.